src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42YFlowRttSpec.groovy

```diff
@@ -0,0 +1,672 @@
+
+    @Shared
+    SwitchTriplet swTEp1OrEp2IsYPoint
+
+    def setupSpec() {
+        swTShEpIsYPoint = topologyHelper.findSwitchTripletWithSharedEpInTheMiddleOfTheChainServer42Support()
+        swTEp1OrEp2IsYPoint = topologyHelper.findSwitchTripletWithSharedEpEp1Ep2InChainServer42Support()
+    }
+
+    @Tags(TOPOLOGY_DEPENDENT)
+    def "Create an Y-Flow (#description) with server42 Rtt feature and check datapoints in tsdb"() {
+        given: "Three active switches with server42 connected"
+        assumeTrue((topology.getActiveServer42Switches().size() >= 3), "Unable to find active server42")
+
+        def swT = isShEpYPoint ? swTShEpIsYPoint : swTEp1OrEp2IsYPoint
+        assert swT, "There is no switch triplet for the further Y-Flow creation"
+
+        when: "Set server42FlowRtt toggle to true"
+        def flowRttToggleInitialState = northbound.featureToggles.server42FlowRtt
+        !flowRttToggleInitialState && northbound.toggleFeature(FeatureTogglesDto.builder().server42FlowRtt(true).build())
+        switchHelper.waitForS42SwRulesSetup()
+
+        and: "server42FlowRtt is enabled on all switches"
+        def initialSwitchesProps = [swT.shared, swT.ep1, swT.ep2].collectEntries { sw -> [sw, switchHelper.setServer42FlowRttForSwitch(sw, true, true)] }
+
+        and: "Create a Y-Flow"
+        YFlowBuilder yFlowRequestParams = setupRequiredParams(yFlowFactory.getBuilder(swT))
+        def yFlow = yFlowRequestParams.build().waitForBeingInState(FlowState.UP)
+        assert isShEpYPoint ? yFlow.sharedEndpoint.switchId == yFlow.yPoint : yFlow.sharedEndpoint.switchId != yFlow.yPoint
+
+        then: "Check if stats for FORWARD and REVERSE directions are available for the first sub-flow"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT + SERVER42_STATS_LAG, 1) {
+            def subFlow1Stats = flowStats.of(yFlow.subFlows.first().flowId)
+            assert subFlow1Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValues()
+            assert subFlow1Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValues()
+        }
+
+        and: "Check if stats for FORWARD and REVERSE directions are available for the second sub-flow"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT + SERVER42_STATS_LAG, 1) {
+            def subFlow2Stats = flowStats.of(yFlow.subFlows.last().flowId)
+            assert subFlow2Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValues()
+            assert subFlow2Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValues()
+        }
+
+        cleanup: "Revert system to original state"
+        yFlow && yFlow.delete() && switchHelper.verifyAbsenceOfServer42FlowRttRules(initialSwitchesProps.keySet())
+        flowRttToggleInitialState != null && northbound.toggleFeature(FeatureTogglesDto.builder().server42FlowRtt(flowRttToggleInitialState).build())
+        initialSwitchesProps && switchHelper.revertToOriginSwitchSetup(initialSwitchesProps, flowRttToggleInitialState)
```

Comment:
This cleanup is already covered in switchelper.setServer42FlowRttForSwitch()

Reply:
yeah, it was an old PR, but now we have new cleanup approach.